### ISSUE TITLE
HOC in fixtures doesn't work anymore

### ISSUE THREAD

**Issue by sergeydt** [react-cosmos/react-cosmos#548]
### What's up?

After version 3.0.0 the following fixture doesn't work anymore:

``` js
import {HOC} from 'myhoc';
import MyComponent from '../MyComponent.jsx';

export default {
  component: HOC(MyComponent)
};

```

Where HOC is Higher Order Component

In version 3.0.0 works fine.

P.S. the HOC in my case is `material-ui` Theme provider (component wrapper).

### Okay, tell me more...
I found that this bug came somewhere from impossibility of extracting `filePath` (or fileName) of such component.

**Comment by skidding** [comment]
Hi @sergeydt,

Just to clarify, does this work in 3.0 but not latest (3.1.1) or it only works in 2.x?

**Comment by sergeydt** [comment]
Hi @skidding ,

Yes, with 3.0.0 it works fine.

When I run `cosmos` through npm script on 3.0.0 < version < 3.1.1 I got the following error:
``` bash
(node:97764) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: Cannot read property 'split' of undefined
(node:97764) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

and with 
``` js
process.on('unhandledRejection', (reason, p) => {
  console.log('Unhandled Rejection at: Promise', p, 'reason:', reason);
});
```

I got this trace
``` bash
[Cosmos] Using webpack config found at config/webpack.config.dev
[Cosmos] Serving static files from public
[Cosmos] See you at http://localhost:8989/
Unhandled Rejection at: Promise Promise {
  <rejected> TypeError: Cannot read property 'split' of undefined
    at module.exports (/Users/sergeydt/work/temp/edc-web-sdk/node_modules/commondir/index.js:25:17)
    at getCommonComponentsDir (/Users/sergeydt/work/temp/edc-web-sdk/node_modules/react-cosmos/lib/server/embed-modules-webpack-loader.js:172:34)
    at Object._callee$ (/Users/sergeydt/work/temp/edc-web-sdk/node_modules/react-cosmos/lib/server/embed-modules-webpack-loader.js:125:35)
    at tryCatch (/Users/sergeydt/work/temp/edc-web-sdk/node_modules/regenerator-runtime/runtime.js:62:40)
    at Generator.invoke [as _invoke] (/Users/sergeydt/work/temp/edc-web-sdk/node_modules/regenerator-runtime/runtime.js:296:22)
    at Generator.prototype.(anonymous function) [as next] (/Users/sergeydt/work/temp/edc-web-sdk/node_modules/regenerator-runtime/runtime.js:114:21)
    at step (/Users/sergeydt/work/temp/edc-web-sdk/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
    at /Users/sergeydt/work/temp/edc-web-sdk/node_modules/babel-runtime/helpers/asyncToGenerator.js:28:13
    at <anonymous> } reason: TypeError: Cannot read property 'split' of undefined
    at module.exports (/Users/sergeydt/work/temp/edc-web-sdk/node_modules/commondir/index.js:25:17)
    at getCommonComponentsDir (/Users/sergeydt/work/temp/edc-web-sdk/node_modules/react-cosmos/lib/server/embed-modules-webpack-loader.js:172:34)
    at Object._callee$ (/Users/sergeydt/work/temp/edc-web-sdk/node_modules/react-cosmos/lib/server/embed-modules-webpack-loader.js:125:35)
    at tryCatch (/Users/sergeydt/work/temp/edc-web-sdk/node_modules/regenerator-runtime/runtime.js:62:40)
    at Generator.invoke [as _invoke] (/Users/sergeydt/work/temp/edc-web-sdk/node_modules/regenerator-runtime/runtime.js:296:22)
    at Generator.prototype.(anonymous function) [as next] (/Users/sergeydt/work/temp/edc-web-sdk/node_modules/regenerator-runtime/runtime.js:114:21)
    at step (/Users/sergeydt/work/temp/edc-web-sdk/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
    at /Users/sergeydt/work/temp/edc-web-sdk/node_modules/babel-runtime/helpers/asyncToGenerator.js:28:13
    at <anonymous>
```


Have just checked and it seems has been fixed in recent release `v3.1.1`. So I think the issue should be closed.. (still anyone who contributes here may consider such usage option to not brake things in future releases). Thanks!

**Comment by skidding** [comment]
@sergeydt sorry you had to go through all of this. I think #545 is the PR that fixed this.

Let us know if you experience any other issue in the future!
